src/solvers/dgmulti/dg.jl

```diff
@@ -136,12 +141,8 @@ function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys, ValueT
     return boundary_conditions
 end
 
-# Allocate nested array type for DGMulti solution storage.
 function allocate_nested_array(uEltype, nvars, array_dimensions, dg)
-    # store components as separate arrays, combine via StructArrays
-    return StructArray{SVector{nvars, uEltype}}(ntuple(_ -> zeros(uEltype,
-                                                                  array_dimensions...),
-                                                       nvars))
```

Comment:
Out of curiosity, what's the motivation for the change and how does it perform?
The `StructArray` version should be SIMD-friendlier, while the `Matrix{<:SVector}` would be better if SIMD isn't possible.

Reply:
The motivation is due mainly to https://github.com/trixi-framework/Trixi.jl/issues/1789. This impacts my specific Trixi solver `DGMulti` and locks Trixi.jl to a specific version of OrdinaryDiffEq. I looked at the array wrappers in RecursiveArrayTools.jl to try to address this, but it takes a lot more energy for me to customize things broadcasting over custom arrays than just use `Base.Array`. I'm also concerned about changes like https://github.com/SciML/RecursiveArrayTools.jl/issues/343. 

My plan is to swap between `Matrix{<:SVector{N, T}}` and `Array{T, 3}` using `unsafe_wrap`, which is currently what Trixi does for other solvers too. My guess is that it won't be nearly as performant once we make these changes, since we have a few pieces of code which are accelerated by Octavian.jl and LoopVec.jl.

However, I'm less worried about efficiency at the moment and more about maintainability, which I think this would help with.